### DISCUSSION TITLE
Compose snapshots with keyboard open

### DISCUSSION THREAD

**Discussion by hrcronin** [cashapp/paparazzi#1133] [Q&A] [answered]
I've tried a few things, but really what I'm looking to do is capture the state of my composable when the keyboard is open:
```
@Composable
fun MyComposable(autoFocus: Boolean) {
   val focusRequester = remember { FocusRequester() }
   LaunchedEffect("auto-focus") {
      if (autoFocus) {
          focusRequester.requestFocus()
      }
   }
   TextField(
       modifier = Modifier.focusRequester(focusRequester)
       // other details here
   )
}

// test implementation
    @Test
    fun keyboardOpen() {
        val composable: @Composable () -> Unit = {
            Column(modifier = Modifier.fillMaxSize()) {
                MyComposable(true)
            }
        }
       // ?? how to get it to capture the keyboard being open?
        paparazzi.snapshot {
            composable()
        }
    }
```

Every time I capture the screenshot I see the cursor in the png, but the keyboard is still shown as hidden. Any ideas on how to capture this state?

**Comment by geoff-powell** [comment, marked as answer]
You could try and capture a video APNG using `paparazzi.gif`. We did make a change for pending recompositions in the last 1.3.5 release to re-render on the first frame for snapshots which might also fix your issue.
